### PR TITLE
Cleanup functional tests / Structure them like libraries tests

### PR DESCRIPTION
**Pull request by steveisok**
You should now be able to run the functional tests locally the same way as running the libraries tests.

Example:
`./dotnet.sh build /t:Test src/tests/FunctionalTests/wasm/Interpreter/console/ /p:TargetOS=Browser /p:TargetArchitecture=wasm /p:Configuration=Release`

### CODE REVIEW

File: src/tests/FunctionalTests/wasm/AOT/browser/Wasm.Aot.Browser.Test.csproj [hunk 1]
```diff
@@ -10,6 +10,16 @@
...
-    <WasmExtraFilesToDeploy Include="index.html" />
   </ItemGroup>
+
+  <Target Name="AfterWasmBuildApp" AfterTargets="WasmBuildApp">
+    <Copy SourceFiles="$(OutDir)\index.html" DestinationFolder="$(WasmAppDir)" />
+  </Target>
+
+  <Target Name="SetMainAssemblyPath" BeforeTargets="Publish">
+    <PropertyGroup>
+      <WasmMainAssemblyPath>$(PublishDir)Wasm.Interpreter.Browser.Test.dll</WasmMainAssemblyPath>
```
Comment (steveisok) [inline src/tests/FunctionalTests/wasm/AOT/browser/Wasm.Aot.Browser.Test.csproj:21]: Change this to `Wasm.Aot.Browser.Test.dll`.

Comment (MaximLipnin) [reply src/tests/FunctionalTests/wasm/AOT/browser/Wasm.Aot.Browser.Test.csproj:21]: yeah, copy-paste, thanks

**Review by MaximLipnin — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]
Just a couple questions, but looks great otherwise!

File: eng/testing/tests.targets [hunk 2]
```diff
@@ -96,8 +96,8 @@
       <RunTestsCommand Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
                                   $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '$(NETCoreAppCurrentVersion)'))">$(RunTestsCommand) --runtime-path "$(TestHostRootPath.TrimEnd('\/'))"</RunTestsCommand>
       <RunTestsCommand Condition="'$(TestRspFile)' != '' and '$(RuntimeFlavor)' != 'Mono'">$(RunTestsCommand) --rsp-file "$(TestRspFile)"</RunTestsCommand>
-      <RunTestsCommand Condition="'$(TargetsMobile)' == 'true'">"$(RunScriptOutputPath)" $(AssemblyName) $(TargetArchitecture) $(TargetOS) $(TestProjectName)</RunTestsCommand>
-      <RunTestsCommand Condition="'$(TargetOS)' == 'Browser'">"$(RunScriptOutputPath)" $(JSEngine) $(AssemblyName).dll $(_withoutCategories.Replace(';', ' -notrait category='))</RunTestsCommand>
+      <RunTestsCommand Condition="'$(TargetsMobile)' == 'true'">"$(RunScriptOutputPath)" $(AssemblyName) $(TargetArchitecture) $(TargetOS) $(TestProjectName) $(ExpectedExitCode)</RunTestsCommand>
+      <RunTestsCommand Condition="'$(TargetOS)' == 'Browser'">"$(RunScriptOutputPath)" $(JSEngine) $(AssemblyName).dll $(Scenario)</RunTestsCommand>
```
[the remark this answers is not in the document]
Comment (steveisok) [reply eng/testing/tests.targets:100]: Correct

File: src/libraries/tests.proj [hunk 3]
```diff
@@ -174,8 +174,10 @@
   <ItemGroup Condition="'$(ArchiveTests)' == 'true' and ('$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'tvOS')">
     <ProjectReference Include="$(MonoProjectRoot)netcore\sample\iOS\Program.csproj"
                       BuildInParallel="false" />
+                      <!-- TODO: remove Exclude after "tests.mobile.targets(107,5): error : Unable to compile method" fixed -->
```
[the remark this answers is not in the document]
Comment (steveisok) [reply src/libraries/tests.proj:177]: @MaximLipnin ?

File: src/tests/FunctionalTests/iOS/Simulator/AOT/Program.cs [hunk 4]
```diff
@@ -6,49 +6,21 @@
...
-// To be replaced with NativeCallableAttribute
-public class MonoPInvokeCallbackAttribute : Attribute
-{
-    public MonoPInvokeCallbackAttribute(Type delegateType) { }
-}
-
 public static class Program
 {
-    // Defined in main.m
+    // to be wired once https://github.com/dotnet/xharness/pull/46 is merged
```
[the remark this answers is not in the document]
Comment (steveisok) [reply src/tests/FunctionalTests/iOS/Simulator/AOT/Program.cs:11]: @MaximLipnin ?

File: src/libraries/tests.proj (hunk 3 above)
[the remark this answers is not in the document]
Comment (MaximLipnin) [reply src/libraries/tests.proj:177]: yes, I filed https://github.com/dotnet/runtime/issues/46984

File: src/tests/FunctionalTests/iOS/Simulator/AOT/Program.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (MaximLipnin) [reply src/tests/FunctionalTests/iOS/Simulator/AOT/Program.cs:11]: Not sure what exactly should be done here.

[the remark this answers is not in the document]
Comment (steveisok) [reply src/tests/FunctionalTests/iOS/Simulator/AOT/Program.cs:11]: Was the comment just copied?  If so, perhaps just get rid of it.

[the remark this answers is not in the document]
Comment (MaximLipnin) [reply src/tests/FunctionalTests/iOS/Simulator/AOT/Program.cs]: Removed the copied comment

File: src/tests/FunctionalTests/iOS/Simulator/Interpreter/Program.cs [hunk 5]
```diff
@@ -6,49 +6,21 @@
...
-// To be replaced with NativeCallableAttribute
-public class MonoPInvokeCallbackAttribute : Attribute
-{
-    public MonoPInvokeCallbackAttribute(Type delegateType) { }
-}
-
 public static class Program
 {
-    // Defined in main.m
+    // to be wired once https://github.com/dotnet/xharness/pull/46 is merged
```
[the remark this answers is not in the document]
Comment (MaximLipnin) [reply src/tests/FunctionalTests/iOS/Simulator/Interpreter/Program.cs]: Removed the copied comment

**Pull request merged**